### ISSUE TITLE
AWS Provider 3.35.0 Breaks this module

### ISSUE THREAD

**Issue by razorsedge** [tmknom/terraform-aws-ssm-parameter#24]
```
module "ssm-ps-creds" {
  source      = "git::https://github.com/tmknom/terraform-aws-ssm-parameter.git?ref=tags/2.0.0"

  name        = "/${var.environment}/${terraform.workspace}/admin"
  value       = "{\"username\": \"fake\", \"password\": \"fake\"}"
  description = "credentials"
  key_id      = var.kms_key_arn
  overwrite   = false
}
```
```
$ terraform plan

Error: expected length of allowed_pattern to be in the range (1 - 1024), got

  on .terraform/modules/ssm-ps-creds/main.tf line 46, in resource "aws_ssm_parameter" "default":
  46:   allowed_pattern = var.allowed_pattern

$ terraform version
Terraform v0.13.6
+ provider registry.terraform.io/hashicorp/aws v3.35.0
```
This works fine with AWS provider 3.34.0.

**Comment by razorsedge** [comment]
https://github.com/hashicorp/terraform-provider-aws/pull/17830 appears to be the culprit.

**Comment by denisbr** [comment]
Introducing validation is good, adding the wrong rule is not so good 🤦 
Broke a bunch of our pipelines due to demanding > 0 length of description.

**Comment by razorsedge** [comment]
This is fixed in [AWS provider 3.36.0](https://github.com/hashicorp/terraform-provider-aws/blob/v3.36.0/CHANGELOG.md) ([#18588](https://github.com/hashicorp/terraform-provider-aws/pull/18588)).
